cypress/support/step_definitions/testSetup.js

```diff
@@ -55,6 +55,23 @@ Given('the environment has been set up for {string} tests', (test) => {
     if (Cypress.env(CONSTANTS.TEST_TYPE).includes('rpc-Only')) {
       Cypress.env(CONSTANTS.IS_RPC_ONLY, true);
     }
+    // fetch device details dynamically
+    if (Cypress.env(CONSTANTS.FETCH_DEVICE_DETAILS_DYNAMICALLY_FLAG)) {
+      if (CONSTANTS.DYNAMIC_DEVICE_DETAILS_MODULES.includes(Cypress.env(CONSTANTS.TEST_TYPE))) {
+        cy.getDeviceData(CONSTANTS.DEVICE_ID, {}, CONSTANTS.ACTION_CORE.toLowerCase()).then(
+          (response) => {
+            if (response) {
+              const method = CONSTANTS.REQUEST_OVERRIDE_CALLS.FETCHDEVICEDETAILS;
+              const requestMap = {
+                method: method,
+                params: response,
+              };
+              cy.sendMessagetoPlatforms(requestMap);
+            }
+          }
+        );
+      }
```

Comment:
can we have comments here explaining about each if else here ?  
what if a config module has Cypress.env(CONSTANTS.FETCH_DEVICE_DETAILS_DYNAMICALLY_FLAG)) == true and the testype == device and there is no override for FETCHDEVICEDETAILS ? 


Did we handle that ?  Also this should be inside try /catch block and proper error  thrown

Reply:
also, what if fetching device data dynamically is a failure? how is that handled? The ticket description has that point mentioned, 

If some error occurs while fetching the source of truth, the fetching status should be updated as FAIL. Then during validation, we would need to add appropriate log mentioning that source of truth fetching failed. Hence using the data key as is.

Reply:
Error handling is done in configModule and configHelper module where all the cases, if token fetching, data fetching anything fails, error is thrown with proper messages